src/engine/runtime.js

```diff
@@ -1219,10 +1219,24 @@ class Runtime extends EventEmitter {
         // Sanitize the placeholder to ensure valid XML
         placeholder = placeholder.replace(/[<"&]/, '_');
 
-        const argJSON = {
-            type: 'input_value',
-            name: placeholder
-        };
+        const iconURI = context.blockInfo.arguments[placeholder].iconURI;
+        let argJSON;
+
+        if (iconURI) {
+            argJSON = {
+                type: 'field_image',
+                name: placeholder,
+                src: iconURI,
+                width: 28,
+                height: 28,
+                alt: context.blockInfo.arguments[placeholder].defaultValue
```

Comment:
Similar to my comment above about the general approach from an extension spec design perspective, I don't think defaultValue is the appropriate analogy to use for the image alt text because it has a different meaning for actual value input arguments.